src/main/java/org/spongepowered/api/entity/player/Player.java

```diff
@@ -144,4 +144,17 @@
      * @return The connection
      */
     PlayerConnection getConnection();
+
+    /**
+     * Kicks the player.
+     */
+    void kick();
+
+    /**
+     * Kicks the player given a reason.
+     *
+     * @param reason The reason for the kick
+     */
+    void kick(String reason);
```

Comment:
I should add on that the kick reason is wrapped in a ChatComponentText, so it doesn't really make sense to have Message